kiwi/xml_description.py

```diff
@@ -52,9 +54,9 @@ class XMLDescription(object):
         path to base XML description file
 
     """
-    def __init__(self, description, derived_from=None):
+    def __init__(self, description=None, derived_from=None, content=None):
         self.description_xslt_processed = NamedTemporaryFile()
-        self.description = description
+        self.description = content and BytesIO(bytes(content)) or description
```

Comment:
What do you want to accomplish here? I see three problems

* hard to read because Boolean circuits instead of using Python
* `content` over `description` (why can't you simply not use description instead of content? or if content is a better name, why not to refactor and change the name of description to content?)
* type mangling, why force it to byte instead of using string? Some python 3 issue? Why not to fix the type issue in the caller and not in the constructor?